File: dev/backend/Vetheria.Vtedy.ApiService/Dto/CommentDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Vetheria.Vtedy.ApiService.Dto
{
    public abstract class CommentDto
    {
        public string Id { get; set; }
        public int UserAccountId { get; set; }
        public string Content { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
    }


    public class ProjectCommentDto : CommentDto
    {
        public string ProjectId { get; set; }
    }


    public class TodoitemCommentDto : CommentDto
    {
        public string TodoitemId { get; set; }
    }
}

File: dev/web/src/app/shared/dto/comment.dto.ts
import { UserAccountDto } from './user-account.dto';

export class CommentCreateRequestDto {
    content: string;
}


export class CommentDto {
    todoitemId: string;
    id: string;
    userAccountId: number;
    content: string;
    createdDate: Date;
    modifiedDate?: any;

    createdBy: UserAccountDto;
}
